### PR TITLE
addTransaction and getCandidates modified

### PR DESCRIPTION
**Pull request by julianlen**

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java [hunk 1]
```diff
@@ -137,40 +148,74 @@ private Optional<Short> getAvailableBucketWithLessUsedGas(long txGasLimit) {
...
             return getSequentialBucket();
         }
 
-        // read - written
-        getCandidatesByReadWriteKeyCollisions(newReadKeys, bucketCandidates);
 
-        if (bucketCandidates.size() > 1) {
-            return getSequentialBucket();
+        // read - written
+        Optional<Short> bucketCandidate = bucketCandidates;
```
Comment (javicid) [inline rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java:159]: Is this really needed?

File: rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java [hunk 2]
```diff
@@ -104,9 +104,16 @@ private void addNewKeysToMaps(RskAddress sender, Short bucketId, Set<ByteArrayWr
...
+
+        if (bucketBySender.containsKey(sender)) {
+
+            if (isSequentialId(bucketId)) {
+                bucketBySender.put(sender, bucketId);
+            }
+
+        } else {
+            bucketBySender.put(sender, bucketId);
+        }
```
[the remark this answers is not in the document]
Comment (julianlen) [reply rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java:116]: If the Map already contains `sender` as a key and its value is the sequentialId, no changes should be made.

[the remark this answers is not in the document]
Comment (julianlen) [reply rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java:116]: Fixed!

```
        if (isSequentialId(bucketId) || !bucketBySender.containsKey(sender)) {
            bucketBySender.put(sender, bucketId);
        }

File: rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java (hunk 1 above)
Comment (julianlen) [reply rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java:159]: Nope, my bad. It stayed there after the refactor.

**Pull request merged**